.github/workflows/build.yml

```diff
@@ -1,18 +1,27 @@
 name: Build
 on: [push, pull_request]
+
 jobs:
   build:
     strategy:
       matrix:
-        java: [17-jdk, 19-jdk]
-    runs-on: ubuntu-22.04
-    container:
-      image: eclipse-temurin:${{ matrix.java }}
-      options: --user root
+        java: [ 17, 19 ]
+        os: [ ubuntu-22.04 ]
+    runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v3
-      - run: ./gradlew build publish --stacktrace --warning-mode fail
-        env:
-          SNAPSHOTS_URL: ${{ secrets.SNAPSHOTS_URL }}
-          SNAPSHOTS_USERNAME: ${{ secrets.SNAPSHOTS_USERNAME }}
-          SNAPSHOTS_PASSWORD: ${{ secrets.SNAPSHOTS_PASSWORD }}
+      - name: checkout repository
+        uses: actions/checkout@v2
+      - name: setup jdk ${{ matrix.java }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+      - name: validate gradle wrapper
+        uses: gradle/wrapper-validation-action@v1
+      - name: build
+        run: ./gradlew build
+      - name: capture build artifacts
+        if: ${{ matrix.java == '17' }} # Only upload artifacts built from latest java on one OS
+        uses: actions/upload-artifact@v2
+        with:
+          name: Artifacts
+          path: enigma-swing/build/libs/
```

Comment:
Snapshot publishing was removed

Reply:
Hm, it does seem like that a fair amount of this could use a bit of fix-up, since the build.yml replacement reverted versioning a fair amount.